XIPs/xip-49-decentralized-backend.md

```diff
@@ -0,0 +1,1099 @@
+   * @notice only callable by admin.
+   */
+  function pause();
+
+  /**
+   * Unpauses the contract.
+   * @notice only callable by admin.
+   */
+  function unpause();
+
+  /**
+   * Upgrades the contract to a new implementation.
+   * @notice only callable by admin.
+   */
+  function upgradeToAndCall(address newImplementation, bytes data);
+
+}
+```
+
+## 3. Specification: Broadcast network
+
+The broadcast network is designed for low latency propagation of messages to all nodes on the network. Although it is impossible to prevent individual nodes from censoring data, the network is designed to make censorship provably detectable and easily circumventable. Totally consistent message ordering is not required in the broadcast network, but should behave well in ordinary cases and be difficult for nodes to manipulate. The network should be eventually consistent: discrepancies between nodes should be resolved over time.
+
+### 3.1. Network structure
+
+In the initial version of the decentralized network, all nodes will hold all of the data (non-sharded). Clients may choose to publish to any node, and read from any node, making message censorship easy to detect as well as to circumvent.
+
+Because latency is critical for a messaging protocol, we choose to adopt a small network size with a broadcast topology: all updates to a given node are immediately broadcast to all other nodes. By reducing the number of hops required for a message to proliferate we can reduce latency and complexity while increasing reliability.
+
+With this size and topology, quality of nodes becomes important (stability, performance), as well as operational capability of node operators. In the testnet phase, we employ a node selection and removal process emphasizing high-quality, diverse, and trusted organizations, with future iterations of the selection process becoming permissionless.
+
+![Diagram illustrating message replication within the broadcast network](./XIPs/assets/xip-49/message-replication.png)
+
+The purpose of this topology is to increase decentralization and trust, not scalability. Each node represents a conceptual grouping of multiple servers run by a single organization, and may choose to scale internally using traditional infrastructure techniques such as sharding and replication across geographies, ensuring that increased demand can be met efficiently.
+
+#### 3.1.1. Node registry
+
+The node registry serves as the source of truth for the complete list of nodes that comprise the XMTP network. This is achieved via the NodeRegistry contract on Ethereum L1, which maintains the following information for each node:
+
+| Field | Description | Mutable? | Unique? |
+| --- | --- | --- | --- |
+| **Node ID** | A strictly increasing index for the node entry. | N | Y |
+| **Public key** | A secp256k1 public key controlled by the node, used for signing and identification purposes.
+Format: Uncompressed point with prefix (0x04) [ P || X || Y ], 65 bytes | N | Y |
```

Comment:
I think this breaks the formatting, at least in the preview

Reply:
Thank you for catching this, @richardhuaaa ! Fixing now.